Sources/PostgresNIO/Connection/PostgresConnection+Configuration.swift

```diff
@@ -0,0 +1,318 @@
+            public var database: String?
+
+            public init(username: String, password: String?, database: String?) {
+                self.username = username
+                self.password = password
+                self.database = database
+            }
+        }
+        
+        // MARK: - Connection options
+        
+        /// Describes options affecting how the underlying connection is made.
+        public struct Options {
+            /// A timeout for connection attempts.
+            ///
+            /// Ignored when using a preexisting communcation channel. (See
+            /// ``PostgresConnection/Configuration/Server-swift.struct/establishedChannel(_:tls:)``.)
+            public let connectTimeout: TimeAmount
+            
+            /// The name to use for SNI (Server Name Indication) when TLS is enabled.
+            ///
+            /// > When set to `nil`:
+            /// If the connection is made to a server over TCP using
+            /// ``PostgresConnection/Configuration/Server-swift.struct/init(host:port:tls:)``, the given `host` is used,
+            /// unless it was an IP address string. If it _was_ an IP, or the connection is made by any other method,
+            /// SNI is disabled.
+            ///
+            /// - Warning: This name is not validated in any way; in particular, there is no attempt to check that it
+            ///   matches the server's TLS certificate, nor is it in any way impacted by the validation setting in the
+            ///   TLS configuration (see ``PostgresConnection/Configuration/Server-swift.struct/TLS-swift.struct``).
+            public let tlsServerName: String?
+            
+            /// Whether the connection is required to provide backend key data (internal Postgres stuff).
+            ///
+            /// This property is provided for compatibility with Amazon RDS Proxy, which requires it to be `false`.
+            /// If you are not using Amazon RDS Proxy, you should probably leave this set to `true` (the default).
+            public let requireBackendKeyData: Bool
+            
+            /// Configure various options for a connection.
+            ///
+            /// Most users should not need to adjust the defaults.
+            ///
+            /// - Parameters:
+            ///   - connectTimeout: See ``connectTimeout``. Defaults to 10 seconds.
+            ///   - tlsServerName: See ``tlsServerName``. Default is `nil`.
+            ///   - requireBackendKeyData: See ``requireBackendKeyData``. Default is `true`.
+            public init(
+                connectTimeout: TimeAmount = .seconds(10),
+                tlsServerName: String? = nil,
+                requireBackendKeyData: Bool = true
+            ) {
+                self.connectTimeout = connectTimeout
+                self.tlsServerName = tlsServerName
+                self.requireBackendKeyData = requireBackendKeyData
+            }
```

Comment:
can we have just an init() that sets the default values? If users want to change this they can use the properties. This makes adding more properties to this way simpler in the future.